contrib/platform/test/com/sun/jna/platform/win32/OleaccTest.java

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2021 Mo Beigi, All Rights Reserved
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.win32;
+
+import com.sun.jna.Memory;
+import com.sun.jna.platform.win32.COM.Accessible;
+import com.sun.jna.platform.win32.COM.COMException;
+import com.sun.jna.platform.win32.COM.IAccessible;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.PointerByReference;
+import com.sun.jna.platform.win32.WTypes.LPSTR;
+import com.sun.jna.platform.win32.WTypes.LPWSTR;
+import com.sun.jna.platform.win32.WinDef.HWND;
+import com.sun.jna.platform.win32.Guid.REFIID;
+import com.sun.jna.platform.win32.WinNT.HRESULT;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static com.sun.jna.platform.win32.WinError.S_OK;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class OleaccTest
+{
+    @BeforeClass
+    public static void setUp() throws Exception {
+        Runtime.getRuntime().exec("calc");
+        Thread.sleep(1000);
+
+        // Initialize COM for this thread...
+        WinNT.HRESULT hr = Ole32.INSTANCE.CoInitialize(null);
```

Comment:
I suggest to initialize COM for multithreaded access. If I remember correctly apartment threaded needs a spinning messgage loop at least at times:

```
Ole32.INSTANCE.CoInitializeEx(Pointer.NULL, Ole32.COINIT_MULTITHREADED);
```

Reply:
In addition, catch the return value. `S_OK` and `S_FALSE` require uninitializing but `RPC_E_CHANGED_MODE` (possible if it's already initialized elsewhere) does not.

Reply:
Yeah, it's probably fine for a test class. I just know I learned about the initialization the hard way.

You could also consider using the COM in a separate thread using the `com.sun.jna.platform.win32.COM.util.ComThread` class.